addons/mail/controllers/discuss/gif.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route, Controller
+import requests
+import json
+
+
+class DiscussGifController(Controller):
+
+    def request_gifs(self, url):
+        try:
+            response = requests.get(url, timeout=3)
+        except requests.exceptions.RequestException:
+            return False
+        if response.status_code != requests.codes.ok:
+            return False
+        return response
+
+    @route("/discuss/gif/search", type="json", auth="public", cors="*")
+    def search(self, search_term, locale="en", country="US", pos=False):
+        config = request.env["ir.config_parameter"].sudo()
+        api_key = config.get_param("discuss.tenor_api_key")
+        content_filter = config.get_param("discuss.tenor_content_filter")
+        ckey = request.env.cr.dbname
+        tenor_url = config.get_param("discuss.tenor_url")
+        tenor_gif_limit = config.get_param("discuss.tenor_gif_limit")
+        url = f"{tenor_url}/v2/search?q={search_term}&key={api_key}&client_key={ckey}&limit={tenor_gif_limit}&contentfilter={content_filter}&locale={locale}&country={country}&media_filter=tinygif"
```

Comment:
I believe these params need to be url encoded

Reply:
Done